### DISCUSSION TITLE
Why the type of the xray test has to be generic ?

### DISCUSSION THREAD

**Discussion by AnthonyTonda** [Xray-App/xray-junit-extensions#30] [General]
Hi,
I was trying your maven artifact.
I am able to produce the custom report. 
However, when I was imported manually to execution plan, why the type of the xray test has to be generic ?

are you planning in future that the custom report can be imported whatever the type of the xray test ?

Regards
Anthony

**Comment by bitcoder** [comment]
Hi, that's how Xray processes Junit XML report so far as detailed in 

https://docs.getxray.app/display/XRAYCLOUD/Taking+advantage+of+JUnit+XML+reports

Another option  would be to support the Xray Json format instead (as recently release for the xray-testng-extensions project for testng), but that will require some work

**Reply by bitcoder** [reply]
Btw, what is your concrete use case? Can you please elaborate?  Are you aiming to report test results for existing manual tests? Are you aiming to import these results for non existent test but you wish that these will be created as manual test for some reason, if so which one?
Thanks

**Reply by AnthonyTonda** [reply]
here is a workflow 

1) SDET creates a xray test, type manual
2) SDET updates that xray test by adding the steps that any SDET can reproduce manually the test
3) SDET implements the Junit test as it can be automated
4) SDET imports the execution result to a test execution
5) As the xray test in not Generic, the execution result can't imported
         so possible actions
         A) to change the type of the test from manual to generic. But we lost the steps ==> I think this action is not valuable
         B) to create another xray test, link to the previous xray test. But it's not convenient to have two xray tests, one to have the details of the xray test and a second xray test to track historically the execution status

my wish is: 
having only one xray test with steps and be able to import execution result of that xray test ( in a xray test execution )

**Comment by bitcoder** [comment]
Ok thanks, that's pretty clear.
